scripts/ccpp_prebuild.py

```diff
@@ -328,6 +329,19 @@ def collect_physics_subroutines(scheme_files):
     os.chdir(BASEDIR)
     return (success, metadata_request, arguments_request, dependencies_request, schemes_in_files)
 
+def check_schemes_in_suites(arguments, suites):
+    """Check that all schemes that are requested in the suites exist"""
+    success = True
+    logging.info("Checking for existence of schemes in suites ...")
+    for suite in suites:
+        for group in suite.groups:
+            for subcycle in group.subcycles:
+                for scheme_name in subcycle.schemes:
+                    if not scheme_name in arguments.keys():
+                        success = False
```

Comment:
It might be nice to return a list of missing schemes so that the exception can include that list. Success would then be an empty list.

Reply:
This is a good idea, but it's a bit against the rest of `ccpp_prebuild.py`. Every subroutine returns a boolean `success` as first argument. Also, the list of suites that cannot not be parsed (be it because they don't exist or because they have syntax errors) is already generated inside the `parse_suites` routine:
```
...
        success = suite.parse()
        if not success:
            logging.error('Parsing suite definition file {0} failed.'.format(sdf))
...
```